File: Ruby/firstVagrantBox/rubyEx/mathdojo.rb
class MathDojo
  # your code here
  def initialize
    @sum = 0
  end

  def add(*num)
    @sum += num.flatten.reduce(0, :+)
    self
  end
def sub(*num)
  @sum -= num.flatten.reduce(0, :+)
  self
  end

def result
  puts @sum
  end


end

challenge1 = MathDojo.new.add(2).add(5, 1).sub(2).result # => 4
challenge2 = MathDojo.new.add(1).add([3, 5, 7, 8], [2, 4.3, 1.25]).sub([2,3], [1.1, 2.3]).result
#challenge2 = MathDojo.new.add(1).add([3, 5, 7, 8], [2, 4.3, 1.25]).subtract([2,3], [1.1, 2.3]).result # => 23.15

File: Python/MathDojo.py
class mathdojo(object):
    def __init__(self):
        self.total = 0


    def add(self, num):
        if type(num)== tuple:
            print "touple"
            print num[0]
            print num[1]
            self.total = self.total + (num[0] + num[1])
            return self
            print self.total

        else:
            print "adding.."
            self.total = self.total + num
            return self


    def sub(self, num):
        if type(num)== tuple:
            self.total = self.total - (num[0] + num[1])
            print self.total
            return self
        else:
            self.total = self.total - num
            return self



    def result(self):
        print self.total

math=mathdojo()

math.add(100).sub(2).add((1,2)).sub((1,2))
